src/Servers/IIS/IIS/src/Core/IISHttpContext.cs

```diff
@@ -385,8 +385,12 @@ public unsafe void SetResponseHeaders()
                             var headerNameBytes = Encoding.UTF8.GetBytes(headerPair.Key);
                             fixed (byte* pHeaderName = headerNameBytes)
                             {
-                                NativeMethods.HttpResponseSetUnknownHeader(_pInProcessHandler, pHeaderName, pHeaderValue, (ushort)headerValueBytes.Length, fReplace: isFirst);
-                            }
+								if(headerValueBytes.Length == 0){
+									throw new ArgumentException("Header value cannot be an empty string");
+								}else{
```

Comment:
This will need to be reformatted to follow our code style. We use the standard VS style, which would be more like this:

```csharp
if (headerValueBytes.Length == 0)
{
	// ...
}
else
}
	// ...
}
``

Note: spaces around keywords and braces get their own line.